service/src/main/java/bio/terra/workspace/amalgam/tps/TpsApiController.java

```diff
@@ -0,0 +1,57 @@
+package bio.terra.workspace.amalgam.tps;
+
+import bio.terra.common.iam.BearerTokenFactory;
+import bio.terra.workspace.generated.controller.TpsApi;
+import bio.terra.workspace.generated.model.ApiTpsPaoCreateRequest;
+import bio.terra.workspace.generated.model.ApiTpsPaoGetResult;
+import java.util.UUID;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+
+/**
+ * This controller simply extracts the bearer token from the request and calls TpsApiDispatch. That
+ * allows the rest of WSM to call TpsApiDispatch using the API classes. If/when we move TPS into its
+ * own service, we can rewrite TpsApiDispatch to make an ApiClient and call the appropriate endpoint
+ * via the REST API. The rest of WSM will not need to change.
+ */
+@Controller
+public class TpsApiController implements TpsApi {
+  private final BearerTokenFactory bearerTokenFactory;
+  private final HttpServletRequest request;
+  private final TpsApiDispatch tpsApiDispatch;
+
+  @Autowired
+  public TpsApiController(
+      BearerTokenFactory bearerTokenFactory,
+      HttpServletRequest request,
+      TpsApiDispatch tpsApiDispatch) {
+    this.bearerTokenFactory = bearerTokenFactory;
+    this.request = request;
+    this.tpsApiDispatch = tpsApiDispatch;
+  }
+
+  // -- Policy Queries --
+  // TODO: PF-1733 Next step is to add group membership constraint
+
+  // -- Policy Attribute Objects --
+  @Override
+  public ResponseEntity<Void> createPao(ApiTpsPaoCreateRequest body) {
+    tpsApiDispatch.createPao(bearerTokenFactory.from(request), body);
```

Comment:
post-refactor, do we need to do any auth checks in the controller here? 
If these are just thin wrappers around the policy service, maybe not

Reply:
Good question! Right now there are no permission checks. I have two options in the policy design doc for how it might be done. I think in the short-term they probably should go in the `TpsApiDispatch` code. If/when we split TPS out, we would move that part into the TPS controller. That leaves the TpsService out of the permission checking business which seems correct.